frontend/src/components/recommendation/prompt/InputBar.tsx

```diff
@@ -1,64 +1,99 @@
 'use client';
-
-import AutoAwesomeIcon from '@mui/icons-material/AutoAwesome';
+import React, { useEffect, useState } from 'react';
+import { InputAdornment, TextField, Button } from '@mui/material';
 import SearchIcon from '@mui/icons-material/Search';
-import { InputAdornment, TextField } from '@mui/material';
 import useTranslation from 'next-translate/useTranslation';
-import type { Dispatch, SetStateAction } from 'react';
+import { GOOGLE_MAPS_API_KEY } from '@/libs/envValues';
+
+const API_KEY = GOOGLE_MAPS_API_KEY;
 
 type Props = {
   placeInput: string;
-  setPlaceInput: Dispatch<SetStateAction<string>>;
+  setPlaceInput: React.Dispatch<React.SetStateAction<string>>;
   handleOpenModal: () => void;
 };
-// ... (other imports)
 
 const InputBar = ({ placeInput, setPlaceInput, handleOpenModal }: Props) => {
   const { t } = useTranslation('home');
+  const [, setAutocomplete] = useState<google.maps.places.Autocomplete | null>(null);
+
+  // Function to format the address as per your requirements
+  const formatShortAddress = (place: google.maps.places.PlaceResult): string => {
+    const city = place.address_components.find((component) => component.types.includes('locality'))
+      ?.long_name;
+
+    const prefecture = place.address_components.find((component) =>
+      component.types.includes('administrative_area_level_1'),
+    )?.long_name;
+
+    const country = place.address_components.find((component) =>
+      component.types.includes('country'),
+    )?.long_name;
+
+    // Customize the address format as needed
+    return `${city}, ${prefecture}, ${country}`;
+  };
+
+  useEffect(() => {
+    const script = document.createElement('script');
+    script.src = `https://maps.googleapis.com/maps/api/js?key=${API_KEY}&libraries=places`;
+    script.async = true;
+    script.onload = () => {
+      const input = document.getElementById('search') as HTMLInputElement;
+      const options = {
+        types: ['establishment'],
+      };
+      const autoCompleteInstance = new google.maps.places.Autocomplete(input, options);
+      setAutocomplete(autoCompleteInstance);
+      autoCompleteInstance.addListener('place_changed', () => {
+        const place = autoCompleteInstance.getPlace();
+        if (place.geometry && place.geometry.location) {
+          // Shorten the address (e.g., take only the first two lines)
+          const shortenedAddress = formatShortAddress(place);
+          setPlaceInput(shortenedAddress);
+        } else {
+          console.error('Error retrieving place details:', place);
+        }
+      });
+    };
+    document.head.appendChild(script);
+    return () => {
+      // Cleanup the script tag to avoid memory leaks
+      document.head.removeChild(script);
+    };
```

Comment:
Your code is React anti-pattern by trying to manipulate dom directly. 
Can I update your code later after you fix the search result?